lib/proxy.js

```diff
@@ -94,9 +94,9 @@ function buildStream (host, options, secure) {
 ////////////////////////////////////////////////////////////
 
 function buildResultingMiddleware (host, options, stream) {
-  console.log(stream)
   return function proxyMiddleware (req, res, next) {
 
+    var _stream = _.clone(stream, true);
```

Comment:
I think this is why you weren't getting the pre and post calls. Since you were modifying stream on every request.